### PR TITLE
docs: improve onError docs by specifying what the error handler is

### LINKED ISSUE #5353: customErrorHandler is a mystery in the documentation
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

In the documentation for [onError](https://fastify.dev/docs/latest/Reference/Hooks/#onerror), it mentions:

> This hook will be executed only after the customErrorHandler has been executed, and only if the customErrorHandler sends an error back to the user (Note that the default customErrorHandler always sends the error back to the user).

The only other mention I can find of customErrorHandler is in the TypeScript reference, which [says the same thing](https://fastify.dev/docs/latest/Reference/TypeScript/#fastifyonerrorhookhandlerrawserver-rawrequest-rawreply-requestgeneric-contextconfigrequest-fastifyrequest-reply-fastifyreply-error-fastifyerror-done---void-promiseunknown--void)

I think that what both of these are referring to is the [argument to setErrorHandler](https://fastify.dev/docs/latest/Reference/Errors/#errors-in-fastify-lifecycle-hooks-and-a-custom-error-handler), which is called "Custom Error Handler", but this is a little surprisingly difficult to figure out, without a link or a searchable name.

### PR DESCRIPTION
**Pull request by tmcw**
- Fixes #5353 

The docs refer to `customErrorHandler` but don't say what it is - this links the `setErrorHandler` method.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Can you please fix the docs linter?

File: docs/Reference/Hooks.md [hunk 1]
```diff
@@ -189,9 +189,10 @@ specific header in case of error.
 It is not intended for changing the error, and calling `reply.send` will throw
 an exception.
 
-This hook will be executed only after the `customErrorHandler` has been
-executed, and only if the `customErrorHandler` sends an error back to the user
-*(Note that the default `customErrorHandler` always sends the error back to the
+This hook will be executed only after
+the [Custom Error Handler set by `setErrorHandler`](./Server.md#seterrorhandler)
+has been executed, and only if the custom error handler sends an error back to the user
```
Comment (Uzlopak) [inline docs/Reference/Hooks.md:194]:
```suggestion
has been executed, and only if the custom error handler sends an error back to the
user
```

**Review by mcollina — APPROVED** [review]
still lgtm

**Comment by Treasuremathew1** [comment]
Hi

**Pull request merged**